perf(linter): replace `phf_set` with `array` in `utils/vitest`

shulaoda opened the pull request:
Related to #10076 

Benchmark repo: https://github.com/shulaoda/rust-benchmark-test
Benchmark result: https://codspeed.io/shulaoda/rust-benchmark-test/benchmarks

BTW, the `VALID_VITEST_FN_CALL_CHAINS` is really huge. I plan to try optimizing it when I have some time later.

Boshen commented:
### Merge activity

* **Apr 16, 12:21 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Apr 16, 12:21 AM EDT**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

shulaoda committed:
perf(linter): replace `phf_set` with `array` in `utils/vitest` (#10427)

Related to #10076

Benchmark repo: https://github.com/shulaoda/rust-benchmark-test
Benchmark result: https://codspeed.io/shulaoda/rust-benchmark-test/benchmarks

BTW, the `VALID_VITEST_FN_CALL_CHAINS` is really huge. I plan to try optimizing it when I have some time later.
